Doubly_Linked_List/Doubly_Linked_List.c

```diff
@@ -0,0 +1,287 @@
+    }
+    return start;
+}
+
+// Function for Displaying Doubly Linked List
+struct node *display(struct node *start)
+{
+    struct node *ptr;
+    ptr = start;
+    while(ptr != NULL)
+    {
+        printf("\t %d", ptr -> data);
+        ptr = ptr -> next;
+    }
+    return start;
+}
+
+// Function for inserting node at beginning
+struct node *insert_beg(struct node *start)
+{
+    struct node *new_node;
+    int num;
+    printf("\n Enter the data : ");
+    scanf("%d", &num);
+    new_node = (struct node *)malloc(sizeof(struct node));
+    new_node -> data = num;
+    start -> prev = new_node;
+    new_node -> next = start;
+    new_node -> prev = NULL;
+    start = new_node;
+    return start;
+}
+
+// Function for inserting node at the end
+struct node *insert_end(struct node *start)
+{
+    struct node *ptr, *new_node;
+    int num;
+    printf("\n Enter the data : ");
+    scanf("%d", &num);
+    new_node = (struct node *)malloc(sizeof(struct node));
+    new_node -> data = num;
+    ptr = start;
+    while(ptr -> next != NULL)
+    {
+        ptr = ptr -> next;
+    }
+    ptr -> next = new_node;
+    new_node -> prev = ptr;
+    new_node -> next = NULL;
+    return start;
+}
+
+// Function for inserting node before the value
+struct node *insert_before(struct node *start)
+{
+    struct node *new_node, *ptr;
+    int num, val;
+    printf("\n Enter the data : ");
+    scanf("%d", &num);
+    printf("\n Enter the value before which the data has to be inserted:");
+    scanf("%d", &val);
+    new_node = (struct node *)malloc(sizeof(struct node));
+    new_node -> data = num;
+    ptr = start;
+    while(ptr -> data != val)
+    {
+        ptr = ptr -> next;
+    }
+    new_node -> next = ptr;
+    new_node -> prev = ptr-> prev;
+    ptr -> prev -> next = new_node;
+    ptr -> prev = new_node;
+    return start;
+}
+
+// Function for inserting node after the value
+struct node *insert_after(struct node *start)
+{
+    struct node *new_node, *ptr;
+    int num, val;
+    printf("\n Enter the data : ");
+    scanf("%d", &num);
+    printf("\n Enter the value after which the data has to be inserted:");
+    scanf("%d", &val);
+    new_node = (struct node *)malloc(sizeof(struct node));
+    new_node -> data = num;
+    ptr = start;
+    while(ptr -> data != val)
+     ptr = ptr -> next;
+    new_node -> prev = ptr;
+    new_node -> next = ptr -> next;
+    ptr -> next -> prev = new_node;
+    ptr -> next = new_node;
+    return start;
+}
+
+// Function for deleting node from beginning
+struct node *delete_beg(struct node *start)
+{
+    struct node *ptr;
+    ptr = start;
+    start = start -> next;
+    start -> prev = NULL;
+    free(ptr);
+    return start;
+}
+
+// Function for deleting node from end
+struct node *delete_end(struct node *start)
+{
+    struct node *ptr;
+    ptr = start;
+    while(ptr -> next != NULL)
+    {
+        ptr = ptr -> next;
+    }
+    ptr -> prev -> next = NULL;
+    free(ptr);
+    return start;
```

Comment:
These are all pointers so what's the need to return them? when the original chain will be modified. You are just returning the address you are receiving as an argument.

Reply:
Yes you are right. Do I need to change it? Or it will work ? While compiling there is no error.

Reply:
It would be great if you can change them.

Reply:
As of now it is difficult for me to change. I am still learning about pointers and data structures.